components/brave_rewards/browser/net/network_delegate_helper.cc

```diff
@@ -129,19 +109,16 @@ int OnBeforeURLRequest(
   std::shared_ptr<brave::BraveRequestInfo> ctx) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::IO);
 
-  if (IsMediaLink(ctx->request_url,
-                  ctx->request->site_for_cookies(),
-                  GURL(ctx->request->referrer()))) {
+  if (IsMediaLink(ctx->request_url, ctx->tab_origin, ctx->referrer)) {
     std::string post_data;
     if (GetPostData(ctx->request, &post_data)) {
-      int render_process_id, render_frame_id, frame_tree_node_id;
-      GetRenderFrameInfo(ctx->request, &render_frame_id, &render_process_id,
-          &frame_tree_node_id);
       base::PostTaskWithTraits(FROM_HERE, {content::BrowserThread::UI},
           base::BindOnce(&DispatchOnUI,
               post_data,
-              ctx->request_url, ctx->request->site_for_cookies(), ctx->request->referrer(),
-              render_process_id, render_frame_id, frame_tree_node_id));
+              ctx->request_url, ctx->tab_url,
+              ctx->referrer.spec(),
+              ctx->render_process_id, ctx->render_frame_id,
```

Comment:
have these always been here and we just weren't using them? Where do they get set?

Reply:
also lint nit: can you put each param on its own line for consistency here?

Reply:
yeah, always. These are set in `FillCTX`. Formatting done